tasks/profilingTasks.ts

```diff
@@ -0,0 +1,112 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import fs from 'fs';
+import * as gulp from 'gulp';
+import * as path from 'path';
+import { basicSlnTestProject, runIntegrationTest } from './testHelpers';
+import { outPath } from './projectPaths';
+import { execSync } from 'child_process';
+
+createProfilingTasks();
+
+function createProfilingTasks() {
+    const profilingOutputFolder = path.join(outPath, 'profiling');
+    gulp.task(`profiling:csharp:${basicSlnTestProject}`, async () => {
+        // Ensure the profiling output folder exists, otherwise the outputs will not get written.
+        fs.mkdirSync(path.join(outPath, 'profiling'), { recursive: true });
+
+        await runIntegrationTest(
+            basicSlnTestProject,
+            path.join('lsptoolshost', 'integrationTests'),
+            `[C#][${basicSlnTestProject}]`,
+            undefined,
+            undefined,
+            {
+                ROSLYN_DOTNET_EventPipeOutputPath: path.join(profilingOutputFolder, 'csharp-trace.{pid}.nettrace'),
+            }
+        );
+
+        const files = fs.readdirSync(profilingOutputFolder);
+        const nettraceFiles = files.filter((f) => f.endsWith('.nettrace'));
+        if (nettraceFiles.length === 0) {
+            throw new Error('No .nettrace files found in the profiling output folder.');
+        }
+    });
+
+    gulp.task('mergeTraces', async () => {
+        await mergeTraces(profilingOutputFolder);
+    });
+
+    gulp.task('profiling', gulp.series(`profiling:csharp:${basicSlnTestProject}`, 'mergeTraces'));
+}
+
+async function mergeTraces(profilingOutputFolder: string) {
+    const files = fs.readdirSync(profilingOutputFolder);
+    const nettraceFiles = files.filter((f) => f.endsWith('.nettrace'));
+    if (nettraceFiles.length === 0) {
+        throw new Error('No .nettrace files found in the profiling output folder.');
+    }
+
+    // Function to spawn a tool
+    function spawnTool(command: string, args: string[], warnOnError = false) {
```

Comment:
this is the same as https://devdiv.visualstudio.com/DevDiv/_git/vs-green?path=/azure-pipelines/profiling/mibc.yml&version=GBmain&_a=contents&line=95&lineStyle=plain&lineEnd=96&lineStartColumn=1&lineEndColumn=1 (internal) but converted from powershell